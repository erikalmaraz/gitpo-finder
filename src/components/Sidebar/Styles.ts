import styled from "styled-components";

export const MainSidebarWrapper = styled.div`
  width: 30%;
  @media screen and (max-width: 768px) {
    width: 100%;
    text-align: center;
    margin-bottom: 20px;
  }
`;

export const ProfileTopContainer = styled.div`
  margin-top: 15px;
  h3 {
    margin-top: 15px;
  }
`;

export const Username = styled.p`
  margin: 2px 0;
  font-size: 19px;
`;

export const ProfilePicture = styled.img`
  max-width: 260px;
  border-radius: 100%;
  object-fit: cover;
  width: 100%;
`;
