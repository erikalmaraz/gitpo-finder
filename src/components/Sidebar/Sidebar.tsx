import React from "react";
import * as S from "./Styles";
import SidebarType from "../../interfaces/sidebar.interface";

const Sidebar = ({ name, image, description, username }: SidebarType) => {
  return (
    <React.Fragment>
      <S.MainSidebarWrapper>
        <S.ProfileTopContainer>
          <div>
            <S.ProfilePicture src={image} alt="Profile picture" />
          </div>
          <div>
            <h3>{name}</h3>
            <S.Username>{username}</S.Username>
          </div>
        </S.ProfileTopContainer>
        <div>{description}</div>
        <div>Location</div>
      </S.MainSidebarWrapper>
    </React.Fragment>
  );
};

export default Sidebar;
