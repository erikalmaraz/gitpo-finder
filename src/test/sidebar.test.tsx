import React from "react";
import { render, screen } from "@testing-library/react";
import Sidebar from "../components/Sidebar/Sidebar";
import renderer from "react-test-renderer";

it("Should render Sidebar correctly", () => {
  const props = {
    name: "Erik Almaraz",
    image: "https://avatars.githubusercontent.com/u/12221606?v=4",
    description: "Description test",
    username: "greatelse",
  };
  const tree = renderer.create(<Sidebar {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
