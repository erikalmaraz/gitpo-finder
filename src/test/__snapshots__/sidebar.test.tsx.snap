// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Sidebar correctly 1`] = `
<div
  className="sc-bdfBQB HHiyw"
>
  <div
    className="sc-gsTEea iFHbBD"
  >
    <div>
      <img
        alt="Profile picture"
        className="sc-hKgJUU hZFgcO"
        src="https://avatars.githubusercontent.com/u/12221606?v=4"
      />
    </div>
    <div>
      <h3>
        Erik Almaraz
      </h3>
      <p
        className="sc-dlfnuX ikmlYA"
      >
        greatelse
      </p>
    </div>
  </div>
  <div>
    Description test
  </div>
  <div>
    Location
  </div>
</div>
`;

exports[`Should render UserFinder correctly 1`] = `
Array [
  <div>
    <input
      className="sc-bdfBQB kVKBGV"
      onChange={[Function]}
      placeholder="Type a username..."
      type="text"
    />
  </div>,
  <div>
    <button
      className="sc-gsTEea ckPYpT"
      onClick={[Function]}
      type="submit"
    >
      Go!
    </button>
  </div>,
]
`;
